src/libfetchers/git.cc

```diff
@@ -143,6 +143,69 @@ struct WorkdirInfo
     bool hasHead = false;
 };
 
+std::vector<PublicKey> getPublicKeys(const Attrs & attrs) {
+    std::vector<PublicKey> publicKeys;
+    if (attrs.contains("publicKeys")) {
+        nlohmann::json publicKeysJson = nlohmann::json::parse(getStrAttr(attrs, "publicKeys"));
+        ensureType(publicKeysJson, nlohmann::json::value_t::array);
+        publicKeys = publicKeysJson.get<std::vector<PublicKey>>();
+    }
+    else {
+        publicKeys = {};
+    }
+    if (attrs.contains("publicKey"))
+        publicKeys.push_back(PublicKey{maybeGetStrAttr(attrs, "keytype").value_or("ssh-ed25519"),getStrAttr(attrs, "publicKey")});
+    return publicKeys;
+}
+
+void doCommitVerification(const Path repoDir, const Path gitDir, const std::string rev, const std::vector<PublicKey>& publicKeys) {
+    // Create ad-hoc allowedSignersFile and populate it with publicKeys
+    auto allowedSignersFile = createTempFile().second;
+    std::string allowedSigners;
+    for (const PublicKey& k : publicKeys) {
+        if (k.type != "ssh-dsa"
+            && k.type != "ssh-ecdsa"
+            && k.type != "ssh-ecdsa-sk"
+            && k.type != "ssh-ed25519"
+            && k.type != "ssh-ed25519-sk"
+            && k.type != "ssh-rsa")
+            warn("Unknow keytype: %s\n"
+                 "Please use one of\n"
+                 "- ssh-dsa\n"
+                 "- ssh-ecdsa\n"
+                 "- ssh-ecdsa-sk\n"
+                 "- ssh-ed25519\n"
+                 "- ssh-ed25519-sk\n"
+                 "- ssh-rsa", k.type);
```

Comment:
```suggestion
                 "- ssh-dsa\n"
                 "  ssh-ecdsa\n"
                 "  ssh-ecdsa-sk\n"
                 "  ssh-ed25519\n"
                 "  ssh-ed25519-sk\n"
                 "  ssh-rsa", k.type);
```

To make absolutely clear that the `-` does not belong.